Dockerfile

```diff
@@ -0,0 +1,70 @@
+# syntax = docker/dockerfile:1
+ARG NODE_VERSION=20.0.0
+ARG SECRET_KEY
+ARG APP_DEBUG
+ARG TEST_SALEOR_API_URL
+FROM node:${NODE_VERSION}-slim as base
+
+# Adjust NODE_VERSION as desired
+
+ENV NODE_VERSION="$NODE_VERSION"
+
+
+LABEL fly_launch_runtime="Next.js"
+
+# Next.js app lives here
+WORKDIR /app
+
+# Install pnpm
+ARG PNPM_VERSION=8.12.0
+RUN npm install -g pnpm@$PNPM_VERSION
+
+
+# Throw-away build stage to reduce size of final image
+FROM base as build
+ARG SECRET_KEY
+ARG APP_DEBUG
+ARG TEST_SALEOR_API_URL
+ARG UPSTASH_URL
+ARG UPSTASH_TOKEN
+ARG APL
```

Comment:
These are runtime variables and server side only. IIRC Next.js should already support runtime env variables (not being compiled on build) already.

Otherwise there is a risk that secrets will leak with a built image

Reply:
I've removed them, and only kept around the ones needed to run `generate:schema`